numpy/f2py/cfuncs.py

```diff
@@ -1045,17 +1045,23 @@
 static int create_cb_arglist(PyObject* fun,PyTupleObject* xa,const int maxnofargs,const int nofoptargs,int *nofargs,PyTupleObject **args,const char *errmess) {
     PyObject *tmp = NULL;
     PyObject *tmp_fun = NULL;
-    int tot,opt,ext,siz,i,di=0;
+    int tot,opt,ext,siz,i,has_self=0;
     CFUNCSMESS(\"create_cb_arglist\\n\");
     tot=opt=ext=siz=0;
     /* Get the total number of arguments */
     if (PyFunction_Check(fun))
         tmp_fun = fun;
     else {
-        di = 1;
+        has_self = ! PyCFunction_Check(fun);
         if (PyObject_HasAttrString(fun,\"im_func\")) {
             tmp_fun = PyObject_GetAttrString(fun,\"im_func\");
         }
+        else if (PyFortran_Check(fun) || PyFortran_Check1(fun)) {
+            tot = maxnofargs;
+            if (xa != NULL)
+                tot += PyTuple_Size((PyObject *)xa);
+            tmp_fun = fun;
+        }
         else if (PyObject_HasAttrString(fun,\"__call__\")) {
             tmp = PyObject_GetAttrString(fun,\"__call__\");
             if (PyObject_HasAttrString(tmp,\"im_func\"))
```

Comment:
This was renamed to `__func__` in python 3 - so is almost certainly wrong here, but I have no idea what it's even testing.